strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/LayoutProviderRegistry.java

```diff
@@ -31,14 +34,18 @@
     @Inject
     private ConfigurationManagementService configurationManagementService;
 
+    @Inject
+    private List<LayoutProvider> layoutProviders;
+
     public LayoutProviderRegistry()
     {
     }
-    
+
     @Override
     @PostConstruct
     public void initialize()
     {
+        layoutProviders.stream().forEach(lp -> addProvider(lp.getAlias(), lp));
```

Comment:
Uhm... Why are we doing it this way?
I thought that every layour provider was getting itself registered in the registy from within each of the provider's `register` method...?

Reply:
In `StrongboxBooter` you use `LayoutProviderRegistry` to initialize repositories. How are you sure that all layout providers have already been registered when we want to use `LayoutProviderRegistry` ? This situation happen in your branch in assembly. Reposiitories couldn't have been initialized properly because `LayoutProviderRegistry` was empty. Layout providers have been registered later. If you don't like this solution, you need to do it in some other way.